sunpy/map/mapbase.py

```diff
@@ -1437,13 +1445,53 @@ def cmap(self):
         """
         Return the `matplotlib.colors.Colormap` instance this map uses.
         """
-        cmap = self.plot_settings['cmap']
-        if isinstance(cmap, str):
-            cmap = plt.get_cmap(cmap)
-            # Set the colormap to be this specific instance so we are not
-            # returning a copy
-            self.plot_settings['cmap'] = cmap
-        return cmap
+        return self._cmap
+
+    @cmap.setter
+    def cmap(self, n):
+        if isinstance(n, str):
+            self._cmap = plt.get_cmap(n)
+        else:
+            self._cmap = n
+        self.plot_settings['cmap'] = self._cmap
+
+    @property
+    def norm(self):
+        return self._norm
+
+    @norm.setter
+    def norm(self, n):
+        if n is None:
+            if self.dtype == np.uint8:
+                self._norm = None
+            else:
+                # Put import here to reduce sunpy.map import time
+                from matplotlib import colors
+                self._norm = colors.Normalize()
+        else:
+            self._norm = n
+        self.plot_settings['norm'] = self._norm
```

Comment:
Remove this line. See note in `cmap` setter above. Sorry again for going back on what I said before 😄